File: 2016/01 No Time for Taxicab/solution.py
dirs = [1j, 1, -1j, -1]
rots = {"R": 1, "L": -1}

dir = 0
coords = 0
directions = open("input1.txt").read().replace(" ", "").split(",")
for where in directions:
    dir = (dir + rots[where[0]]) % 4
    coords += dirs[dir] * int(where[1:])
    print(coords)
print(abs(coords.real) + abs(coords.imag))

visited = {0}
coords = 0
dir = 0
directions = open("input1.txt").read().replace(" ", "").split(",")
twc = False
for where in directions:
    dir = (dir + rots[where[0]]) % 4
    for i in range(int(where[1:])):
        coords += dirs[dir]
        #print(where, coords)
        if coords in visited and not twc:
            print(coords)
            twc = True
        visited.add(coords)
print(abs(coords.real) + abs(coords.imag))

# 191 --- too high

File: 2016/01 No Time for Taxicab/solution.lua
dirs_x = {0, 1, 0, -1}
dirs_y = {1, 0, -1, 0}
dir = 0
x, y = 0, 0

f = io.open("input1.txt")
s = f:read()
for rot, num in string.gmatch(s, "([RL])(%d+)") do
    if rot == "R" then
        dir = dir + 1
    else
        dir = dir - 1
    end
    dir = dir % 4
    dist = tonumber(num)
    x = x + dirs_x[dir + 1] * dist
    y = y + dirs_y[dir + 1] * dist
end
print(x, y, x + y)